pkg/proto/nbcontract/v1/httpproxyconfig.proto

```diff
@@ -10,7 +10,7 @@ message HTTPProxyConfig {
   repeated string https_proxy_urls = 2;
 
   // NoProxy represents a list of endpoints including IPs, domains, or other network CIDRs that should not be accessed through the proxy
-  repeated string no_proxy = 3;
+  repeated string no_proxy_urls = 3;
```

Comment:
NO_PROXY are not URLs

Reply:
But it's NO_PROXY_URLS in the cse_cmd.sh
https://github.com/Azure/AgentBaker/blob/102df0a193cb00042749092c107185873964d7fd/parts/linux/cloud-init/artifacts/cse_cmd.sh#L115

Reply:
Can you check on the actual use in the VHD? Could be a naming "bug" in the current use, that we can fix in the contract ...

Reply:
yep I did. In cse_config.sh, it writes the NO_PROXY_URLS to /etc/systemd/system.conf.d/proxy.conf.
And usually the proxy.conf looks like this,
`{
  "proxies": {
    "http-proxy": "http://proxy.example.com:3128",
    "https-proxy": "https://proxy.example.com:3129",
    "no-proxy": "*.test.example.com,.example.org,127.0.0.0/8"
  }
}`
no-proxy should be a list of URLs

Reply:
Nope, those entries in NO_PROXY are not [URLs](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax). :-) Domains, domain suffixes (with or without regex), IP addresses, CIDR blocks etc. - the full list is complicated - not URLs.

Reply:
I see. Didn't realize the difference. I will use no_proxy_entries. Does this sound good to you?
Will update other scripts as well when we modify them later.

Reply:
Yes, `no_proxy_entries` would work - as long as we decide to break them into parts, which I think is worth trying. (Otherwise I would just keep it as a single `no_proxy` string ... `http_proxy`, `https_proxy` and `no_proxy` are well-known environment variable names ...)

Which brings me to this (and apologies for not noticing it earlier): for `http_proxy` and `https_proxy`, neither "repeated", nor "_urls" (plural) suffix make sense. It needs to be a single URL for each (unless I am missing something major). So either just `http_proxy` and `https_proxy` (preferred, IMO), or `http_proxy_url` and `https_proxy_url`. In either case, not repeated.

Reply:
That's a good catch! It doesn't make sense to have multiple http proxy. I changed the contract variable names to `http_proxy` and `https_proxy`. Will further update the bootstrap scripts later on.